### PR TITLE
JCF: when you build a plugin, don't allow it to have functions which …

### LINKED ISSUE #100: Building a plugin with undefined function calls doesn't cause an error
This was discovered by Eric in his work on listrev. Basically, if you look at commit `92a64096eb` on the listrev branch `eflumerf/RequestBasedListrev`, in the `ListReverser` DAQ module we have `get_next_generator` declared, and even called by another function in the module, but `get_next_generator` is never defined. Regardless, compilation proceeds without an error. This needs to be fixed.

### PR DESCRIPTION
**Pull request by jcfreeman2**
…aren't defined

### CODE REVIEW

**Comment by jcfreeman2** [comment]
Also merge https://github.com/DUNE-DAQ/appfwk/pull/266 once this has been merged, as that repo has a couple of plugins with undefined functions.

**Comment by jcfreeman2** [comment]
Changed the linker option from `--no-undefined` to `--no-allow-shlib-undefined`. While the latter is stricter, the stack continues to build:
https://github.com/DUNE-DAQ/daq-release/actions/runs/6787216401/job/18449532932

**Comment by jcfreeman2** [comment]
Two more changes:
1) `daq_add_plugin` will automatically link against `cetlib` 
2) I reverted from `--no-allow-shlib-undefined` to `--no-undefined`, since `--no-allow-shlib-undefined` didn't cause the error with Eric's listrev code that we wanted and which motivated this Issue

File: cmake/DAQ.cmake [hunk 1]
```diff
@@ -780,8 +784,13 @@ function(daq_add_plugin pluginname plugintype)
   endif()
 
   add_library( ${pluginlibname} MODULE ${PLUGIN_PATH}/${pluginname}.cpp)
+  target_link_options( ${pluginlibname} PRIVATE "LINKER:--no-undefined") # A plugin should have all its contents defined
 
-  target_link_libraries(${pluginlibname} ${PLUGOPTS_LINK_LIBRARIES})
+  if (NOT DEFINED CETLIB OR NOT DEFINED CETLIB_EXCEPT)
```
Comment (alessandrothea) [inline cmake/DAQ.cmake]: Isn't a check on CETLIB sufficient? CETLIB_EXCEPT is a dependency of CETLIB.
I would keep the check minima;.

**Review by eflumerf — APPROVED** [review]
LGTM. We could add a check on the presence of `cetlib` and link it in automatically if defined, but it is equally valid to push it down to the package CMakeLists.txt.

**Pull request merged**